src/LAMMPS.jl

```diff
@@ -139,14 +139,18 @@ mutable struct LMP
                 handle = API.lammps_open_no_mpi(length(args), args, C_NULL)
             end
         end
-
+        check(handle)
         this = new(handle)
-        check(this)
         finalizer(close!, this)
         return this
     end
 end
-Base.unsafe_convert(::Type{Ptr{Cvoid}}, lmp::LMP) = lmp.handle
+
+function Base.unsafe_convert(::Type{Ptr{Cvoid}}, lmp::LMP)    
+    lmp.handle == C_NULL && error("The LMP object doesn't point to a valid LAMMPS instance! "
+            * "This is usually caused by calling `LAMMPS.close!` or through serialization and deserialization.")
+    return lmp.handle
+end
```

Comment:
```suggestion
function Base.cconvert(::Type{Ptr{Cvoid}}, lmp::LMP)    
    lmp.handle == C_NULL && error("The LMP object doesn't point to a valid LAMMPS instance! "
            * "This is usually caused by calling `LAMMPS.close!` or through serialization and deserialization.")
    return lmp
end
Base.unsafe_convert(::Type{Ptr{Cvoid}}, lmp::LMP) = lmp.handle
```

Reply:
I'm not quite sure what difference that would make. Could you explain?

Reply:
The expectation in Julia is that `unsafe_convert` is well `unsafe` it shouldn't perform error checking.
Most notably the invariant that Julia guarantees is that the input argument to `unsafe_convert` remains rooted during the duration of the `ccall`, but the output is not rooted.

For argument conversion (or in our case argument checking) one can use [`cconvert`](https://docs.julialang.org/en/v1/base/c/#Base.cconvert)


```
ccall(:f, Cvoid, (Ptr{Cvoid},), x)
```

turns into:
```
obj = Base.cconvert(Ptr{Cvoid}, x)
GC.@preserve obj begin
     raw_obj = Base.unsafe_convert(Ptr{Cvoid},obj)
     ccall(:f, Cvoid, (Ptr{Cvoid},), raw_obj)
end
```
It can sometimes matter how much code is inside the `GC.@preserve`.